### PR TITLE
Update Message Types, DIDComm Transports and JSON-LD Compatibility RFCs

### PR DESCRIPTION
**Pull request by TimoGlastra**

### CODE REVIEW

File: concepts/0020-message-types/README.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ did:sov:123456789abcdefghi1234;spec/exampleprotocol/1.0/exampletype
 
 This example DID document shows a service endpoint that includes a name property (emphasized) whose purpose is to enable creation of DID references that can deterministically select that service in order to have an algorithmic transformation into a concrete URI.
 
-```json
+```jsonc
 {
   "@context": "https://w3id.org/did/v1",
   "id": "did:example:123456789abcdefghi",
```
Comment (swcurran) [inline concepts/0020-message-types/README.md]: Could yo change this as well? An ID should (in general) be a GUID, not a DID.

**Comment by swcurran** [comment]
All the rest of the changes are good - thanks for this.  If you could add the one additional change, we can merge this.  Or we can merge it now :-).

File: concepts/0020-message-types/README.md (hunk 1 above)
Comment (dhh1128) [reply concepts/0020-message-types/README.md]: @swcurran , this is an `id` in a DID doc, not an `id` in a DIDComm message. Therefore, having the `id` be a DID is actually normal.

**Comment by dhh1128** [comment]
I'm pretty sure that the comment about message-types/README.md should be considered resolved. If you don't agree, @swcurran , let me know, and I'll go fix it myself. In the meantime, I'm merging to keep us moving.

File: concepts/0020-message-types/README.md (hunk 1 above)
Comment (swcurran) [reply concepts/0020-message-types/README.md]: Doh.  Sorry.  Trying to figure out why a DIDDoc would be in that document.  It will be deleted soon :-)

**Pull request merged**